plugin/x/src/interface/server.h

```diff
@@ -67,7 +68,7 @@ class Server {
 
   virtual xpl::Mutex &get_client_exit_mutex() = 0;
 
-  virtual Ssl_context *ssl_context() const = 0;
+  virtual std::shared_ptr<Ssl_context> ssl_context() const = 0;
```

Comment:
It looks like changing the signature of this method is not needed if you are not going to save the context in `xpl::iface::Client::m_ssl`

Reply:
Same answer: the shared_ptr ensures that 2 connections can use 2 different ssl contexts (old and new) at the same time.